### PR TITLE
Update swift 5.2

### PR DESCRIPTION
**Pull request by fdiaz**
Updates `ArgumentParser` and `SwiftSyntax` to be able to be used in Swift 5.2

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -8,6 +8,8 @@ jobs:
     steps:
     # Checks-out the repo. More at: https://github.com/actions/checkout
     - uses: actions/checkout@v2
+    - name: Select Xcode version
```
Comment (bachand) [inline .github/workflows/ci.yml:11]: Maybe we can comment that this can be removed eventually?

File: Package.swift [hunk 2]
```diff
@@ -14,8 +14,8 @@ let package = Package(
     .library(name: "SwiftInspectorCore", targets: ["SwiftInspectorCore"])
   ],
   dependencies: [
-    .package(url: "https://github.com/apple/swift-argument-parser", .upToNextMinor(from: "0.0.1")),
-    .package(url: "https://github.com/apple/swift-syntax.git", .exact("0.50100.0")),
+    .package(url: "https://github.com/apple/swift-argument-parser", .upToNextMinor(from: "0.0.4")),
+    .package(name: "SwiftSyntax", url: "https://github.com/apple/swift-syntax.git", .exact("0.50200.0")),
```
Comment (fdiaz) [inline Package.swift:18]: It's now required that you pass the name for this. CI was failing if we didn't.

File: Sources/SwiftInspectorCore/ImportsAnalyzer.swift [hunk 3]
```diff
@@ -97,7 +97,7 @@ public final class ImportsAnalyzer: Analyzer {
     var submoduleIdentifier: String = ""
 
     for child in syntaxNode.children {
-      guard let accessPath = child as? AccessPathSyntax else {
+      guard let accessPath = child.as(AccessPathSyntax.self) else {
```
Comment (bachand) [inline Sources/SwiftInspectorCore/ImportsAnalyzer.swift:100]: Can you explain this to me?

File: Sources/SwiftInspectorCore/InitializerAnalyzer.swift [hunk 4]
```diff
@@ -36,11 +36,11 @@ public final class InitializerAnalyzer: Analyzer {
     var statements: [InitializerStatement] = []
     let syntax = try cachedSyntaxTree.syntaxTree(for: fileURL)
 
-    var reader = InitializerSyntaxReader(shouldVisitIdentifier: shouldVisit) { [unowned self] node in
+    let reader = InitializerSyntaxReader(shouldVisitIdentifier: shouldVisit) { [unowned self] node in
       let statement = self.initializerStatement(from: node)
       statements.append(statement)
     }
-    syntax.walk(&reader)
+    reader.walk(syntax)
```
Comment (bachand) [inline Sources/SwiftInspectorCore/InitializerAnalyzer.swift:43]: This syntax makes more sense anyway

File: Sources/SwiftInspectorCore/InitializerAnalyzer.swift [hunk 5]
```diff
@@ -192,19 +192,19 @@ private final class InitializerSyntaxReader: SyntaxVisitor {
     self.onNodeVisit = onNodeVisit
   }
 
-  func visit(_ node: ClassDeclSyntax) -> SyntaxVisitorContinueKind {
+  override func visit(_ node: ClassDeclSyntax) -> SyntaxVisitorContinueKind {
```
Comment (bachand) [inline Sources/SwiftInspectorCore/InitializerAnalyzer.swift:195]: So it's a subclass and not a protocol?

**Comment by bachand** [comment]
Any other interesting learnings from the release notes?

File: .github/workflows/ci.yml (hunk 1 above)
Comment (fdiaz) [reply .github/workflows/ci.yml:11]: I think we want to keep being explicit about what Xcode version we're using given the Swift requirement. I hate that we need to do this so hacky, but I didn't find a better way to do it with GitHub actions.

File: Sources/SwiftInspectorCore/ImportsAnalyzer.swift (hunk 3 above)
Comment (fdiaz) [reply Sources/SwiftInspectorCore/ImportsAnalyzer.swift:100]: @bachand how Swiftsyntax structures nodes moved from being protocol based to struct based. You can read more about [this here](https://github.com/apple/swift-syntax/blob/master/Changelog.md#modelling-syntax-nodes-as-structs)

File: Sources/SwiftInspectorCore/InitializerAnalyzer.swift (hunk 5 above)
Comment (fdiaz) [reply Sources/SwiftInspectorCore/InitializerAnalyzer.swift:195]: Yes, this was changed with the new version https://github.com/apple/swift-syntax/blob/master/Changelog.md#swift-52

Comment (bachand) [reply Sources/SwiftInspectorCore/InitializerAnalyzer.swift:195]: Cool release notes are fairly helpful

**Pull request merged**